utils/scarb-macro-interface/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+use std::fmt::Display;
+
+pub mod stable_abi;
+
+#[derive(Debug)]
+#[allow(dead_code)]
+pub enum ProcMacroResult {
+    /// Plugin has not taken any action.
+    Leave,
+    /// Plugin generated TokenStream replacement.
+    Replace(TokenStream),
+    /// Plugin ordered item removal.
+    Remove,
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct TokenStream(String);
+
+impl TokenStream {
+    pub fn new(s: String) -> Self {
```

Comment:
Do you need to make this part of public API facing macro developers?

Reply:
Well, since the plugin can return TokenStream to replace the one he got as an argument, it kind of makes sense to me? 🤔